### PR TITLE
Support installing on PHP 7.0 with some minor sacrifice

### PR DESCRIPTION
**Pull request by sanmai**
This is in addition to what is done in #38. Totally optional, even if I would've make a use of this.

### CODE REVIEW

File: src/BusinessTime.php [hunk 1]
```diff
@@ -265,7 +265,7 @@ public function subBusinessHours(float $businessHoursToSub): self
      * @return int
      */
     public function diffInBusinessDays(
-        ?DateTimeInterface $time = null,
+        DateTimeInterface $time = null,
```
Comment (sanmai) [inline src/BusinessTime.php]: `DateTimeInterface $time = null` is same as `?DateTimeInterface $time = null`

File: src/Deadline/RecurringDeadline.php [hunk 2]
```diff
@@ -121,16 +121,14 @@ public function hasPassedBetween(
     public function firstTimePassedBetween(
         BusinessTime $start,
         BusinessTime $end
-    ): ?BusinessTime {
+    ) {
```
Comment (sanmai) [inline src/Deadline/RecurringDeadline.php]: Return type also shown in PHPDoc, should not cause a problem.

File: tests/Functional/Remote/WebCalFi/WebCalFiFactoryTest.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ public function setUp()
      *
      * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function testUkHolidaysContainsChristmas(): void
+    public function testUkHolidaysContainsChristmas()
```
Comment (sanmai) [inline tests/Functional/Remote/WebCalFi/WebCalFiFactoryTest.php]: Not part of an interface, shouldn't be a problem.

File: tests/Unit/BusinessTime/RoundTest.php [hunk 4]
```diff
@@ -147,9 +147,9 @@ public function roundProvider(): array
      */
     public function testCeil(
         string $time,
-        ?Interval $precision,
-        string $expectedCeilTime
-    ): void {
+        Interval $precision = null,
+        string $expectedCeilTime = ''
+    ) {
```
Comment (sanmai) [inline tests/Unit/BusinessTime/RoundTest.php]: `?Interval $precision` is same as `Interval $precision = null`

File: src/Remote/WebCalFi/WebCalFiConstraint.php [hunk 5]
```diff
@@ -61,12 +61,10 @@ public function narrate(DateTimeInterface $time): string
      *
      * @return WebCalFiDate|null
      */
-    private function dateFor(DateTimeInterface $time): ?WebCalFiDate
+    private function dateFor(DateTimeInterface $time)
```
Comment (sanmai) [inline src/Remote/WebCalFi/WebCalFiConstraint.php]: BC break.

**Comment by hughgrigg** [comment]
Cool. It's a shame to lose the stronger type hinting, but I guess this makes sense to let more people use the library. Thanks for doing this.

**Comment by sanmai** [comment]
No problem. Can we have a release then?

**Comment by hughgrigg** [comment]
Good point!

https://github.com/hughgrigg/php-business-time/releases/tag/1.1.1


Alexey Kopytko <notifications@github.com> 于2018年11月7日周三 下午11:41写道：

**Pull request merged**